internal/cloudapi/v2/openapi.v2.yml

```diff
@@ -735,45 +735,111 @@ components:
-        image_type:
-          $ref: '#/components/schemas/ImageTypes'
         repositories:
           type: array
           items:
             $ref: '#/components/schemas/Repository'
         ostree:
           $ref: '#/components/schemas/OSTree'
-        upload_options:
-          $ref: '#/components/schemas/UploadOptions'
-    ImageTypes:
-      type: string
-      enum:
-        - aws
-        - aws-ha-rhui
-        - aws-rhui
-        - aws-sap-rhui
-        - azure
-        - azure-rhui
-        - edge-commit
-        - edge-container
-        - edge-installer
-        - gcp
-        - gcp-rhui
-        - guest-image
-        - image-installer
-        - iot-commit
-        - iot-container
-        - iot-raw-image
-        - vsphere
+
+    AWSEC2ImageRequest:
+      allOf:
+        - $ref: '#/components/schemas/AbstractImageRequest'
+        - type: object
+          required:
+            - image_type
+          properties:
+            image_type:
+              type: string
+              enum:
+                - aws
+                - aws-ha-rhui
+                - aws-rhui
+                - aws-sap-rhui
+            upload_options:
+              $ref: '#/components/schemas/AWSEC2UploadOptions'
+
+    AWSS3ImageRequest:
+      allOf:
+        - $ref: '#/components/schemas/AbstractImageRequest'
+        - type: object
+          required:
+            - image_type
+          properties:
+            image_type:
+              type: string
+              enum:
+                - edge-commit
+                - edge-installer
+                - guest-image
+                - image-installer
+                - iot-commit
+                - iot-raw-image
+                - vsphere
+            upload_options:
+              $ref: '#/components/schemas/AWSS3UploadOptions'
+
+    GCPImageRequest:
+      allOf:
+        - $ref: '#/components/schemas/AbstractImageRequest'
+        - type: object
+          required:
+            - image_type
+          properties:
+            image_type:
+              type: string
+              enum:
+                - gcp
+                - gcp-rhui
+            upload_options:
+              $ref: '#/components/schemas/GCPUploadOptions'
+
+    AzureImageRequest:
+      allOf:
+        - $ref: '#/components/schemas/AbstractImageRequest'
+        - type: object
+          required:
+            - image_type
+          properties:
+            image_type:
+              type: string
+              enum:
+                - azure
+                - azure-rhui
+            upload_options:
+              $ref: '#/components/schemas/AzureUploadOptions'
+
+    ContainerImageRequest:
+      allOf:
+        - $ref: '#/components/schemas/AbstractImageRequest'
+        - type: object
+          required:
+            - image_type
+          properties:
+            image_type:
+              type: string
+              enum:
+                - edge-container
+                - iot-container
```

Comment:
Are you sure that nobody relies on uploading these also to S3? 🤔  Just checking...